### PR TITLE
Addresses i60, i63 re imports of units and components

### PR DESCRIPTION
**Pull request by kerimoyle**
This is my attempt at rewording sections C01 imports, C02 unit references and C04 component references in order to clarify the issues in #60 and #63.  Please feel free to emend as you see fit!

### CODE REVIEW

**Comment by kerimoyle** [comment]
It's easier to read in the rendered version: See https://cellml-specification-dev.readthedocs.io/en/latest/reference/specC01_interpretation_of_imports.html (nb this references my "dev" site, not the "official" one as it's not merged yet ...)

File: reference/sectionC_interpretation.rst [hunk 1]
```diff
@@ -313,19 +327,27 @@ Component reference
...
-#. It is noted, for the avoidance of doubt, that CellML models MAY apply
-   the previous rule recursively, to reference an :code:`import component`
-   element which in turn references another :code:`import component`
-   element.
+#. It is noted, for the avoidance of doubt, that importing a CellML
+   :code:`component` element SHALL additionally and recursively import:
+
+      #. Any and all encapsulated :code:`component` children of the imported
+         :code:`component`, and
+      #. Any and all :code:`units` elements referenced by the imported
```
Comment (MichaelClerx) [inline reference/sectionC_interpretation.rst]: Still needs to make clear that:
 - if I import component X, which uses units Y, that does _not_ mean I can now refer to Y in my new model
 - if I import component X, which uses units Y, and I import units Y, that _does_ let me refer to Y
 - if I import component X, which uses units Y, and I import units Y, that does not give me to Y's in my model

Comment (MichaelClerx) [reply reference/sectionC_interpretation.rst]: Guess one way to think of it is that any `<import>` imports the entire model, and the individual component and units imports are just namespacing rules that let you refer to the imported stuff from the importing model?

Comment (kerimoyle) [reply reference/sectionC_interpretation.rst]: Sorry, I don't understand the last point? 

> if I import component X, which uses units Y, and I import units Y, **that does not give me to Y's in my model** 

Re namespacing stuff, yeah, kinda, but we need a way to make clear that if you import five components from the same model, you've really imported the entire model five times ...

Comment (kerimoyle) [reply reference/sectionC_interpretation.rst]: @MichaelClerx I think it's clearer now?  But let me know what you think (nb: added section to C1 as well as to C2, C4)

File: reference/sectionA_definitions.rst [hunk 2]
```diff
@@ -324,7 +324,7 @@ The following data representation formats are defined for use in this specificat
 
       4. An exponent SHALL be represented by an exponent separator character, followed by the :ref:`integer string representation<specA_integer>` of the value of the exponent.
       Non-negative exponents MAY begin with the Basic Latin plus sign character :unicode:`002B` as the sign indicator.
-         The exponent separator character SHALL be either the Basic Latin ‘E’ character :unicode:`0045` or the Basic Latin ‘e’ character :unicode:`0065`.
+      The exponent separator character SHALL be either the Basic Latin ‘E’ character :unicode:`0045` or the Basic Latin ‘e’ character :unicode:`0065`.
```
Comment (MichaelClerx) [inline reference/sectionA_definitions.rst]: Change ‘ to "

File: reference/sectionC_interpretation.rst [hunk 3]
```diff
@@ -5,15 +5,21 @@
...
 .. marker_interpretation_of_imports_start
 
 .. _specC_interpretation_of_imports:
 
 Interpretation of imports
 -------------------------
 
-#. Each :code:`import` element present in a :ref:`CellML infoset<specA_cellml_infoset>` (the importing infoset) SHALL define a new and separate instance of the CellML infoset referenced by the :code:`href` attribute (the imported infoset).
-   See :ref:`Units reference<specC_units_reference>` and    :ref:`Component reference<specC_component_reference>` for the specifics    of importing units and components.
+#. Each :code:`import` element present in a :ref:`CellML infoset<specA_cellml_infoset>` (the importing infoset) SHALL define a new and distinct instance of the CellML infoset which is referenced by the :code:`href` attribute (the imported infoset). This imported infoset instance SHALL provide the context in which the child of an import element (a :code:`units` or :code:`component` element) is defined and interpreted.
```
Comment (MichaelClerx) [inline reference/sectionC_interpretation.rst]: Newline before "This"

File: reference/sectionC_interpretation.rst [hunk 4]
```diff
@@ -5,15 +5,21 @@
...
+
+#. The importing of an element SHALL be considered recursively such that items referenced or imported by the :code:`import` element child (a :code:`units` or :code:`component` element) are themselves available to that :code:`import` element child.
```
Comment (MichaelClerx) [inline reference/sectionC_interpretation.rst]: "referenced or imported by" isnt' right. You reference things that are in the model either through definition or importing

File: reference/sectionC_interpretation.rst [hunk 5]
```diff
@@ -5,15 +5,21 @@
...
+   It is noted, for the avoidance of doubt, that the importing infoset SHALL NOT have access to such recursively imported items, but only to the explicitly specified :code:`import` element child.
```
Comment (MichaelClerx) [inline reference/sectionC_interpretation.rst]: `, but only to...` --> `, unless these are explicitly imported.` ?

File: reference/sectionC_interpretation.rst [hunk 6]
```diff
@@ -23,18 +29,20 @@ Interpretation of imports
...
 
-#. A units reference SHALL be a CellML identifier and SHALL be interpreted dependent on the context of the :ref:`CellML model<specA_cellml_model>` in which it occurs, according to the units referencing rules defined later in this section.
-
-#. A :ref:`CellML infoset<specA_cellml_infoset>` MUST NOT contain a units reference for which no referencing rule can be held to have been followed.
+#. A units reference SHALL be a CellML identifier and SHALL be  interpreted dependent on the context of the :ref:`CellML model<specA_cellml_model>` in which it occurs.
 
 #. The units referencing rules are:
 
-   #. Where there is a :code:`units` element with a :code:`name` attribute identical to the units reference, then the units reference SHALL refer to that :code:`units` element.
+   #. Where there is a :code:`units` element with a :code:`name` attribute identical to the units reference within the same infoset, then the units reference SHALL refer to that :code:`units` element.
```
Comment (MichaelClerx) [inline reference/sectionC_interpretation.rst]: Removes spaces at end of line

Comment (MichaelClerx) [reply reference/sectionC_interpretation.rst]: I don't like the "within the same infoset" bit! It's true, but it make the sentence "Where there is THING with FILTER in X", but you could also read it as "Where there is THING with (FILTER IN X)"

File: reference/sectionC_interpretation.rst [hunk 7]
```diff
@@ -23,18 +29,20 @@ Interpretation of imports
...
+   #. Where there is a :code:`units` element with a :code:`name` attribute identical to the units reference within the same infoset, then the units reference SHALL refer to that :code:`units` element.  
 
    #. Where there is an :code:`import units` element in the :ref:`CellML infoset<specA_cellml_infoset>`, such that the :code:`import units` element has a :code:`name` attribute identical to the units reference, then the units reference SHALL be treated with respect to referencing rules as if the units reference appeared in the imported infoset, and referring to the :code:`name` specified in the :code:`units_ref` attribute of the :code:`import units` element.
 
    #. Where the units reference is equal to the value in the ‘Name’ column of the :ref:`Built-in units table<table_built_in_units>`, then the units reference SHALL be a reference to the built-in units corresponding to that row of the table.
 
+#. A :ref:`CellML infoset<specA_cellml_infoset>` MUST NOT contain a units reference for which no referencing rule can be held to have been followed.
```
Comment (MichaelClerx) [inline reference/sectionC_interpretation.rst]: This needs to change (or to go?).
It's left over from where we had a list of criteria to determine which rule was "held to be followed", and then ended saying at least one should be "held to be followed". We reworded all of that (phew!), so this one should change too

File: reference/sectionC_interpretation.rst [hunk 8]
```diff
@@ -236,11 +244,17 @@ Component reference
 
 #. A component reference present in an information item which is a descendant of a :code:`model` element SHALL be identical to either the :code:`name` attribute on a :code:`component` element or to the :code:`name` attribute on an :code:`import component` element.
 
-#. A component reference which is identical to the :code:`name` attribute on a :code:`component` element SHALL be treated as a reference to that :code:`component` element.
+#. A component reference which is identical to the :code:`name` attribute on a :code:`component` element in the same infoset SHALL be treated as a reference to that :code:`component` element.
```
Comment (MichaelClerx) [inline reference/sectionC_interpretation.rst]: Same sentence structure issues as above

File: reference/sectionC_interpretation.rst (hunk 3 above)
Comment (kerimoyle) [reply reference/sectionC_interpretation.rst]: Done.

File: reference/sectionC_interpretation.rst (hunk 4 above)
Comment (kerimoyle) [reply reference/sectionC_interpretation.rst]: Hmm. If I have a model that imports a units item from another model, I want to be able to say that the imported units will drag anything it needs along with it, whether those things are imported or defined in the imported model.  I wanted "referenced" to mean "locally requested by the imported item" ... is there a better term or way of saying that?

File: reference/sectionC_interpretation.rst (hunk 6 above)
Comment (kerimoyle) [reply reference/sectionC_interpretation.rst]: Agreed.  Have changed it to: Where, within the same infoset, there is a :code:`units` element with a :code:`name` attribute identical to the units reference, then the units reference SHALL refer to that :code:`units` element.

File: reference/sectionC_interpretation.rst (hunk 7 above)
Comment (kerimoyle) [reply reference/sectionC_interpretation.rst]: The list of criteria for units referencing are above still?

File: reference/sectionC_interpretation.rst (hunk 8 above)
Comment (kerimoyle) [reply reference/sectionC_interpretation.rst]: Changed as above

File: reference/sectionC_interpretation.rst (hunk 5 above)
Comment (kerimoyle) [reply reference/sectionC_interpretation.rst]: Done (ish) ... I think the sentence is getting too long and icky though ...
It is noted, for the avoidance of doubt, that the importing infoset SHALL NOT have access to such recursively imported items, but only to the explicitly specified :code:`import` element child, unless that child is itself explicitly imported.

**Comment by kerimoyle** [comment]
See #86 instead ... I merged this one by accident, and can't get it back :(

File: reference/sectionC_interpretation.rst (hunk 4 above)
Comment (MichaelClerx) [reply reference/sectionC_interpretation.rst]: Cont. on #86!

File: reference/sectionC_interpretation.rst (hunk 6 above)
Comment (MichaelClerx) [reply reference/sectionC_interpretation.rst]: Thanks!

File: reference/sectionC_interpretation.rst (hunk 5 above)
Comment (MichaelClerx) [reply reference/sectionC_interpretation.rst]: Thank!

**Comment by MichaelClerx** [comment]
Thanks @kerimoyle ! Final two minor points are now in #86

**Pull request merged**